app/src/main/java/com/example/plantsapp/domain/repository/UserRepository.kt

```diff
@@ -6,4 +6,5 @@ interface UserRepository {
     fun setUser(user: User)
     fun requireUser(): User
     fun isAuthorized(): Boolean
+    fun signOut()
```

Comment:
I think from point of UserRepository view it does not know about some logic like "SignIn" or "SignOut" and it will be more clear to create method "clearUser"